Dockerfile

```diff
@@ -1,5 +1,10 @@
 FROM node:18
 WORKDIR /app
+RUN npm i -g @nestjs/cli
+COPY server/package*.json ./
+WORKDIR /app/server
+RUN npm install
 COPY . .
 EXPOSE 3000
-CMD node server/server.js
\ No newline at end of file
+# RUN npm run build
+CMD ["npm", "run", "start:dev", "--prefix", "server"]
```

Comment:
Did you test these changes to your Dockerfile? You made some significant changes here, but your test plan doesn't clarify whether you tested them.

I also suspect that if you deploy to fly.io, you'll encounter an out-of-memory issue since the machines are limited to 200MB (I had the same issue). When you build your Docker image, can you run `docker build --memory=200m [...]` in order to enforce a 200MB memory limit, and see if that works?

Reply:
Yes, I tested these  and they work https://hr-tutorial.fly.dev/, you can check it here. I had memory issues with fly.io, I received through email and I scaled the memory.